.github/workflows/ci-workflow.yml

```diff
@@ -81,14 +94,35 @@ jobs:
           bundle update
           bundle install --jobs 4 --retry 3
         working-directory: demo
+
+      # Cache cypress binary
+      - uses: actions/cache@v2
+        with:
+          path: /home/runner/.cache/Cypress
+          key: cypress-binary-${{ hashFiles('package-lock.json') }}
+
+      # Npm install
       - uses: bahmutov/npm-install@v1
         with:
           working-directory: demo
 
-      # Run a webpack build as a smoke test
+      # Run a webpack build
       - run: ./bin/webpack
         working-directory: demo
 
+      # Run cypress tests
+      - run: ./bin/rails cypress:run
+        working-directory: demo
+        env:
+          CYPRESS_RAILS_CYPRESS_OPTS: '--browser chrome'
+
+      # Archive test screenshots
+      - uses: actions/upload-artifact@v2
+        with:
+          name: cypress-screenshots
+          path: cypress/screenshots
+        if: ${{ failure() }}
```

Comment:
Cypress has automatic screenshot capturing (and video if we want it) so we can upload those on failure.